### ISSUE TITLE
Mentions strange sort with groups

### ISSUE THREAD

**Issue by alfazaz** [julianlam/nodebb-plugin-mentions#117]
Hello !

I don't understand sort of mentions autocomplete list. For example, when I begin to type `@a`

 in composer on nodebb community forum, it gives
![image](https://user-images.githubusercontent.com/9592199/70723062-9544c280-1cf8-11ea-8870-80fe8ec98a25.png)

Is it possible to have a greatest sorting ? If `@a` is typed, only propositions beginning with the `a` letter should be proposed at first (in alphabetical order), for groups and users. No ?

It's annoying if there are many many mentions possibles because the dropdown has only ten items in it... Bad items are showing first and we can't see good items...

**Comment by julianlam** [comment]
You're correct, it's bugged out, and has been for quite some time. What's _supposed_ to happen is a list of matching users and groups are returned. It doesn't seem like the groups are filtered properly (or at all!) :smile:

**Comment by barisusakli** [comment]
Looks like it's matching anything that contains `a` instead of those starting with `a`. Might be intentional.

**Comment by alfazaz** [comment]
If it is intentional, it's not great because of the limitation to ten items in the dropdown...

For example, if I have many groups with close names (promo-00-01, promo-01-02, etc...), we can't see all of these (see https://github.com/julianlam/nodebb-plugin-mentions/issues/118) and there can be unshowed items in dropdown also by a replacement by other terms which don't begin with p letter...

I think the dropdown selector could be improved (see also https://github.com/julianlam/nodebb-plugin-mentions/issues/119).

;-)
